src/test/scala/com/microsoft/hyperspace/index/LogicalPlanSerDeTests.scala

```diff
@@ -69,10 +74,59 @@ class LogicalPlanSerDeTests extends SparkFunSuite with SparkInvolvedSuite {
     StructType(attributes.map(a => StructField(a.name, a.dataType, a.nullable, a.metadata)))
   }
 
-  test("Serde query with Hadoop file system relation.") {
+  test("Serde query with Hadoop file system parquet relation.") {
     verifyPlanSerde(scanNode, "hadoopFsRelation.plan")
   }
 
+  private def verifyUnSerializableFileFormat(fileFormat: FileFormat): Unit = {
+    val relation: HadoopFsRelation =
+      scanNode.relation.asInstanceOf[HadoopFsRelation].copy(fileFormat = fileFormat)(spark)
+    val updatedScanNode = scanNode.copy(relation = relation)
+
+    // This file format is serializable unless isSplittable is called on it. isSplittable api
+    // initializes some unserializable internal objects which break the serialization logic.
+    val kryoSerializer = new KryoSerializer(spark.sparkContext.getConf)
+    KryoSerDeUtils.serialize(kryoSerializer, fileFormat)
+
+    // Confirm that isSplittable makes serialization fail
+    intercept[KryoException] {
+      fileFormat.isSplitable(spark, Map(), new Path("path"))
+      KryoSerDeUtils.serialize(kryoSerializer, fileFormat)
+    }
+
+    // Now verify if Hyperspace serialization still works with this format
+    verifyPlanSerde(updatedScanNode, "hadoopFsRelation.plan")
+  }
+
+  test("Serde query with Hadoop file system csv relation.") {
+    val csvFormat = new CSVFileFormat
+    // CSVFormat is unserializable due to internal unserializable members. Verify as below.
+    verifyUnSerializableFileFormat(csvFormat)
+  }
+
+  test("Serde query with Hadoop file system json relation.") {
+    val jsonFormat = new JsonFileFormat
+    // JsonFileFormat is unserializable due to internal unserializable members. Verify as below.
+    verifyUnSerializableFileFormat(jsonFormat)
+  }
+
+  test("Serde query with Hadoop file system orc relation.") {
+    val orcFormat = new OrcFileFormat
+    val relation: HadoopFsRelation =
+      scanNode.relation.asInstanceOf[HadoopFsRelation].copy(fileFormat = orcFormat)(spark)
+    val orcScanNode = scanNode.copy(relation = relation)
+
+    // Orc file format is serializable as it doesn't contain unserializable objects like in csv or
```

Comment:
you should have this check for orc/parquet? I think you can also update `verifyUnSerializableFileFormat` to take in extra param `expectSerializationError` and reuse it.

Reply:
and rename the function to `verifyFileFormat`.

Reply:
thanks @imback82 , updated as suggested and also added test for parquet.

Reply:
thanks @imback82, done